lib/resty/kafka/client.lua

```diff
@@ -151,7 +175,23 @@ local function _fetch_metadata(self, new_topic)
             end
             self.brokers, self.topic_partitions = brokers, topic_partitions
 
-            return brokers, topic_partitions
+            -- fetch ApiVersions for compatibility
+            local resp, err = bk:send_receive(api_versions_encode(self.client_id))
+            if not resp then
+                ngx_log(INFO, "broker fetch api versions failed, err:", err,
+                          ", host: ", host, ", port: ", port)
+            else
+                local errcode, api_versions = api_versions_decode(resp)
+
+                if not errcode == 0 then
+                    ngx_log(INFO, "broker fetch api versions failed, err:", Errors[err],
+                    ", host: ", host, ", port: ", port)
+                else
+                    self.api_versions = api_versions
+
+                    return brokers, topic_partitions, api_versions
+                end
+            end
```

Comment:
style: better introduce a function `_fetch_api_version` for these lines?

Reply:
Good idea, I will do it.

Reply:
changed